### PR TITLE
Make mockchain UTXO serializable with serde

### PR DESCRIPTION
**Pull request by CodeSandwich**
This will allow Jormungandr to write mockchain UTXO dump to JSON returned from REST API

### CODE REVIEW

File: chain-addr/src/lib.rs [hunk 1]
```diff
@@ -24,11 +24,17 @@ use std::string::ToString;
 // temporary re-use just to define
 use cardano::redeem::{self, PublicKey};
 
-use chain_core::property;
+use chain_core::property::{self, Serialize as PropertySerialize};
```
Comment (mzabaluev) [inline chain-addr/src/lib.rs]: Nitpick: I'd just use `property::Serialize` like everywhere else, the whole-module import is there anyway.

Comment (CodeSandwich) [reply chain-addr/src/lib.rs]: I need to import the trait so I can use its methods on structs implementing it. Without this import I would have to use UFC.

File: chain-addr/src/lib.rs [hunk 2]
```diff
@@ -24,11 +24,17 @@ use std::string::ToString;
...
+
+#[cfg(feature = "generic-serialization")]
+use serde::Serializer as SerdeSerializer;
+#[cfg(feature = "generic-serialization")]
+use serde_derive::Serialize;
 
 // Allow to differentiate between address in
 // production and testing setting, so that
 // one type of address is not used in another setting.
+#[cfg_attr(feature = "generic-serialization", derive(Serialize))]
```
Comment (NicolasDP) [inline chain-addr/src/lib.rs]: we don't need this one

File: chain-addr/src/lib.rs [hunk 3]
```diff
@@ -39,6 +45,7 @@ pub enum Discrimination {
 ///
 /// * Single address : just a single ed25519 spending public key
 /// * Group address : an ed25519 spending public key followed by a group public key used for staking
+#[cfg_attr(feature = "generic-serialization", derive(Serialize))]
```
Comment (NicolasDP) [inline chain-addr/src/lib.rs]: we don't need this one

**Pull request merged**